keycloak/main.tf

```diff
@@ -0,0 +1,102 @@
+
+data "aws_eks_cluster" "this" {
+  name = var.cluster_name
+  
+data "aws_region" "current" {}
+  
+resource "kubernetes_namespace" "this" {
+  depends_on = [
+    var.module_depends_on
+  ]
+  count = var.namespace == "" ? 1 - local.argocd_enabled : 0
+  metadata {
+    name = var.namespace_name
+  }
+}
+
+resource "helm_release" "keyclok" {
+  count = 1 - local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  repository    = local.repository
+  name          = local.name
+  chart         = local.chart
+  chart_version = var.chart_version
+  namespace     = local.namespace
+  recreate_pods = true
+  timeout       = 1200
+  
+    dynamic "set" {
+    for_each = local.conf
+  
+      content {
+      name  = set.key
+      value = set.value
+    }
+}
+ 
+resource "local_file" "this" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content  = yamlencode(local.application)
+  filename = "${path.root}/${var.argocd.path}/${local.name}.yaml"
+}
+
+locals {
+  argocd_enabled = length(var.argocd) > 0 ? 1 : 0
+  namespace      = coalescelist(kubernetes_namespace.this, [{ "metadata" = [{ "name" = var.namespace }] }])[0].metadata[0].name
+
+  repository    = "https://charts.bitnami.com/bitnami"
+  name          = "keyclok"
+  chart         = "keyclok"
+  chart_version = var.chart_version
+  conf          = merge(local.conf_defaults, var.conf)
+  conf_defaults = merge({
```

Comment:
What about another param? Like an admin password or ingress? Just install helm chart not enough for us